docs/source/guide/refactor_code/index.rst

```diff
@@ -1,7 +1,156 @@
 Refactoring Code
 ================
 
-.. toctree::
-   :maxdepth: 1
+LineaPy's artifacts store the complete development history in intelligent graph representation,
+and this enables automatic transformation of the development code.
 
-   ../../tutorials/01_refactor_code
+An important application of such transformation is “program slicing” where the development code is
+“sliced” to only retain minimal necessary operations generating the final state/object of interest
+(e.g., trained model).
+
+Traditionally, a major bottleneck in any data science work is refactoring messy development code into
```

Comment:
I'd add more context. Something like:

"Data science development is characterized by nimble, iterative experimentation and exploratory data analysis. Data scientists explore many possibilities before reaching the final result. The rapid exploration process often leads to long, messy code, over 90% of which has no impact on the final result." 
and then go into the problem of refactoring.